lucene/core/src/test/org/apache/lucene/index/TestIndexWriterMergePolicy.java

```diff
@@ -396,10 +396,12 @@ protected void merge(MergePolicy.OneMerge merge) throws IOException {
             waitForMerge.await();
             writer.updateDocument(new Term("id", "2"), d2);
```

Comment:
This test fully deleted the 2nd segment (since it only had `d2`, which we are deleting here).  Maybe also add a similar case that leaves one not deleted document in the segment too?  I think we optimize 100% deleted merge source segments too, so this should tickle slightly different path.